core/src/ast/lifetimes.rs

```diff
@@ -174,7 +174,7 @@ impl LifetimeEnv {
                 self.extend_implicit_lifetime_bounds(typ, behind_ref);
             }
             TypeName::Option(typ) => self.extend_implicit_lifetime_bounds(typ, None),
-            TypeName::Result(ok, err) => {
+            TypeName::Result(ok, err) | TypeName::DiplomatResult(ok, err) => {
```

Comment:
suggestion: can it just be a boolean flag on the AST node? That was my original plan for this and it makes it cleaner I think

Reply:
Ok